### PR TITLE
Some Refactoring

### PR DESCRIPTION
**Pull request by rexwangcc**
- Bump the version of `cromwell-tools` and refactor the usage accordingly.
- Add some unit tests for `analysis_agent`.
- Attempt to integrate to Travis CI.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+language: python
```
Comment (parthshahva) [inline .travis.yml]: @rexwangcc should we run our tests in gitlab instead?

File: tracker/common/dcp_agents/analysis_agent.py [hunk 2]
```diff
@@ -2,37 +2,43 @@
...
+        """Interacts with the HCA DCP Pipelines Execution Service."""
         deployment = os.environ['DEPLOYMENT_STAGE']
         self.cromwell_url = 'https://cromwell.caas-prod.broadinstitute.org'
         self.cromwell_collection = 'lira-test' if deployment == 'integration' else f'lira-{deployment}'
         gcp_credentials_file_for_analysis = os.environ.get('GCP_ACCOUNT_ANALYSIS_INFO')
-        if gcp_credentials_file_for_analysis:
-            with open(gcp_credentials_file_for_analysis, 'r') as f:
-                service_account_key = json.load(f)
-            self.auth = self._get_auth(service_account_key)
+        self.auth = cromwell_tools.cromwell_auth.CromwellAuth.harmonize_credentials(service_account_key=gcp_credentials_file_for_analysis,
```
Comment (parthshahva) [inline tracker/common/dcp_agents/analysis_agent.py]: Will need to merge in master. This is updated a bit as the secret is now stored in aws secrets manager

File: tracker/common/dcp_agents/tests/test_analysis_agent.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import pytest
```
Comment (parthshahva) [inline tracker/common/dcp_agents/tests/test_analysis_agent.py]: @sampierson will be combining all the dcp agents in dcp, dcp-diag, and this repo into dcplib. He can move these tests when that is done

Comment (rexwangcc) [reply tracker/common/dcp_agents/tests/test_analysis_agent.py]: That sound like a good move. So what do we want to do with these test cases now? Should we take them out or just leave them here and @sampierson could make a PR later to move them?

File: .travis.yml (hunk 1 above)
Comment (rexwangcc) [reply .travis.yml]: Good question, to me it depends on:
- Are we still paying for Travis?
- Do we think it's important to expose test logs to external contributors? (Gitlab is not open to the public)
- Do we want to mirror every repo to Gitlab? I'm not sure if we can run unit tests in Gitlab without having a repo there.
- How many concurrent runners available on Gitlab.

File: tracker/common/dcp_agents/analysis_agent.py (hunk 2 above)
Comment (rexwangcc) [reply tracker/common/dcp_agents/analysis_agent.py]: done!

File: .travis.yml (hunk 1 above)
Comment (parthshahva) [reply .travis.yml]: @rexwangcc I don't think we're paying for Travis but could be wrong. I think gitlab actually allows for readonly access to the builds if configured. This repo will probably already be mirrored there soon for CI/CD. Also, we can work with BrianH to increase # of workers as needed over time

File: tracker/common/dcp_agents/tests/test_analysis_agent.py (hunk 3 above)
Comment (parthshahva) [reply tracker/common/dcp_agents/tests/test_analysis_agent.py]: I think its better to just merge this in and then Sam and copy/modify as needed

**Comment by parthshahva** [comment]
@rexwangcc Looks like the travis builds are failing. Feel free to merge this in after they pass! Will be great to get the first tests into the repo :)

File: .travis.yml (hunk 1 above)
Comment (rexwangcc) [reply .travis.yml]: All make sense!

**Comment by rexwangcc** [comment]
@parthshahva Ah, curse from having a test! 😢 

It's failing at some imports:
```
from tracker.common.config import TrackerInfraConfig
ModuleNotFoundError: No module named 'tracker'
```
It seems the imports cannot find the `tracker` module anymore, I will try to fix it.

**Pull request merged**